### ISSUE TITLE
Editor options have confusing highlighting

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#3733]
## Issue Overview
After clicking on the More icon and switching to Code Editor, I then clicked on the More icon again and based on the highlighting of the options, I have no idea what view I'm in. Am I in the Visual Editor, because it's highlighted and looks like it's selected, or am I in the Code Editor because it has a tick next to it, even though it doesn't appear to be highlighted like the other option.

![screenshot_526](https://user-images.githubusercontent.com/2078245/33416195-58c6e4ac-d5ef-11e7-9c59-1c87bb3ac3d5.png)

## Steps to Reproduce (for bugs)
1. Click on the 3 vertical dot icon in the top right-hand corner when editing content
2. Switch to Code Editor
3. Click on the 3 vertical dot icon again
4.

## Expected Behavior
I should be able to determine at a quick glance, which item is selected.

## Current Behavior
The choice for highlighting the selected option is confusing

## Possible Solution
Don't put a background colour and border around the option which isn't actually selected

## Related Issues and/or PRs
#3708

_Firefox Quantum 57.0
Gutenberg Version 1.8.1
macOS Sierra 10.12.1_

**Comment by jasmussen** [comment]
The first item is a focus highlight to indicate what action is invoked when you press "enter". In my understanding, accessibility guidelines suggest that when a dropdown menu is opened, the first item has focus. 

We've had a discussion in the past on whether this focus should be visible, lingering there, even if you're not using the keyboard to invoke things (see [this codepen for gory details](https://codepen.io/joen/pen/gRYazN)), but the concensus from the WordPress accessibility team _seems_ to be that this is intended behavior. @afercia can you chime in?

**Comment by maddisondesigns** [comment]
I'm all for making things accessible and having focus styles, but the way this is currently styled, it looks like it's actually selected, rather than a menu item that just has focus.

You could easily keep the focus indicator whilst at the same time, not make it look like it's selected, by having a dotted border and removing the background colour.

![screenshot_242](https://user-images.githubusercontent.com/2078245/33512076-6c5e1c18-d77c-11e7-8258-8cd08b09f802.png)

**Comment by jasmussen** [comment]
Here's how that style could look:

![buttons focus states](https://user-images.githubusercontent.com/1204802/33602685-642e9690-d9b0-11e7-99c4-74674d617b4a.png)

I don't dislike that, but I want to make sure with @afercia whether this is accessible or not.

**Comment by afercia** [comment]
The dotted border color seems too light to me, it should have a sufficient color contrast. There's no point in (correctly) using a shape as focus indicator if then the shape can't be easily distinguished from the background color.

A more generic consideration is that the Visual / Code choice would be a perfect case for 2 radio buttons. It's a single choice option: the native control for that are radio buttons 🙂  Not sure reinventing the wheel and using buttons for visual purposes it's so ideal.
Also, there's no semantic indication about which of the 2 buttons is currently selected. Radio buttons would communicate this information natively.

**Comment by maddisondesigns** [comment]
I like the idea of changing that Visual/Code Editor option to a radio button to make it more obvious which one is selected.

Also, if that border colour is too light, or doesn't have enough contrast, how about just changing the colour of it. I still think that just the border alone makes a better option than having the whole background coloured, along with a border, as that makes the option look like it's been selected, rather than just having focus. Of course though, it's pointless if no-one can see it.

Here's the same styles but using a blue border instead. I've used the same blue as the (Update) button.

![screenshot_545](https://user-images.githubusercontent.com/2078245/33638196-fdc17e9e-da78-11e7-9eca-04daff3fdbcf.png)

And here's an alternate option with a Dashed border instead of Dotted.

![screenshot_546](https://user-images.githubusercontent.com/2078245/33638207-0df2241c-da79-11e7-9eae-4d4c56da62d3.png)

**Comment by jasmussen** [comment]
> A more generic consideration is that the Visual / Code choice would be a perfect case for 2 radio buttons. It's a single choice option: the native control for that are radio buttons 🙂 Not sure reinventing the wheel and using buttons for visual purposes it's so ideal.
Also, there's no semantic indication about which of the 2 buttons is currently selected. Radio buttons would communicate this information natively.

That's not a hill I want to die on, so I won't block efforts in that direction. However I'd like to present a quick argument for why they look like they look right now, which is inspired by how menus in Google Docs work (look for line-height).

In my mind, radio groups are visible both before the click, and after the click, essentially the point being that you can see the dot is now in a different inset ring, compounding the _group_ aspect. 

In this case, picking an item in the menu immediately selects it, which in my head makes the checkmark feel more natural. 

Same argument for why it can make sense to use a Switch component instead of a checkbox — the switch saves the change immediately, whereas the checkbox should be accompanied with a "Save Changes" or "Apply" button.

**Comment by jasmussen** [comment]
Here's a version of the focus outlines that uses $gray-dark-900, the darkest we have before black:

![buttons focus states darker](https://user-images.githubusercontent.com/1204802/33652249-fda6d810-da68-11e7-8055-97f9256c0d34.png)

**Comment by afercia** [comment]
> which is inspired by how menus in Google Docs work (look for line-height).

Well, looking at their implementation, the whole line-height menu is an ARIA menu:
`role="menu" aria-haspopup="true"`
and each item is a [menuitemcheckbox](https://www.w3.org/TR/wai-aria/roles#menuitemcheckbox):
`role="menuitemcheckbox" aria-checked="true"`

this way, their checked / unchecked state is reported correctly:

![screenshot 19](https://user-images.githubusercontent.com/1682452/33655120-84ac1218-da72-11e7-9baa-f5b4d362df6b.png)

Apart from the visual presentation, if you really want to build custom controls that are non-native in HTML, then semantics and interaction should be properly rebuilt using ARIA because this is the only way other software can understand what's going on. And this is what Google Docs does.
Native radio buttons would give all of this for free, out of the box. If we're able to build a proper ARIA menu with proper roles and properties, then I'm all for it.

**Comment by maddisondesigns** [comment]
@jasmussen Does #4512 actually address the concerns that I specifically raised in this issue? It doesn't appear to. As of Gutenberg 2.0.0, the way the Visual/Code Editor option is highlighted is still extremely confusing.

**Comment by jasmussen** [comment]
I don't believe 4512 reached 2.0 in time for release. But this is how master looks:

![screen shot 2018-01-20 at 07 02 28](https://user-images.githubusercontent.com/1204802/35180400-f2215b2c-fdaf-11e7-9719-c189e5051862.png)

**Comment by maddisondesigns** [comment]
Thanks for the pic. I'll retest this once a new version of the plugin is released
